pkg/controller/config/match/match.go

```diff
@@ -0,0 +1,65 @@
+package match
+
+import (
+	configv1alpha1 "github.com/open-policy-agent/gatekeeper/apis/config/v1alpha1"
+)
+
+type Operation string
+
+const (
+	Audit   = Operation("audit")
+	Sync    = Operation("sync")
+	Webhook = Operation("webhook")
+	Star    = Operation("*")
+)
+
+type Set struct {
+	ExcludedNamespaces map[Operation][]string
+}
+
+var allOperations = []Operation{
+	Audit,
+	Webhook,
+	Sync,
+}
+
+var configMapSet = &Set{
+	ExcludedNamespaces: make(map[Operation][]string),
+}
+
+func NewSet() *Set {
+	return configMapSet
+}
+
+func (s *Set) Update(entry []configv1alpha1.MatchEntry) {
```

Comment:
We need thread safety primitives for all read/write updates to this object.

Reply:
added